tests/unit/Rules/FilterVarTest.php

```diff
@@ -13,64 +13,57 @@
 
 namespace Respect\Validation\Rules;
 
-use PHPUnit\Framework\TestCase;
+use Respect\Validation\Test\RuleTestCase;
+use const FILTER_FLAG_QUERY_REQUIRED;
+use const FILTER_VALIDATE_BOOLEAN;
+use const FILTER_VALIDATE_EMAIL;
+use const FILTER_VALIDATE_FLOAT;
+use const FILTER_VALIDATE_INT;
+use const FILTER_VALIDATE_URL;
 
 /**
- * @group  rule
+ * @group rule
+ *
  * @covers \Respect\Validation\Rules\FilterVar
- * @covers \Respect\Validation\Exceptions\FilterVarException
+ *
+ * @author Gabriel Caruso <carusogabriel34@gmail.com>
+ * @author Henrique Moody <henriquemoody@gmail.com>
  */
-class FilterVarTest extends TestCase
+final class FilterVarTest extends RuleTestCase
 {
     /**
-     * @expectedException \Respect\Validation\Exceptions\ComponentException
-     * @expectedExceptionMessage Cannot validate without filter flag
-     */
-    public function testShouldThrowsExceptionWhenFilterIsNotDefined(): void
-    {
-        new FilterVar();
-    }
-
-    /**
+     * @test
+     *
      * @expectedException \Respect\Validation\Exceptions\ComponentException
      * @expectedExceptionMessage Cannot accept the given filter
      */
-    public function testShouldThrowsExceptionWhenFilterIsNotValid(): void
+    public function itShouldThrowsExceptionWhenFilterIsNotValid(): void
```

Comment:
Why not move this exception test to phpt test?

Reply:
Because this exception is a condition of the constructor, not a failure in the validation.